### DISCUSSION TITLE
New flake8 pyproject.toml plugin

### DISCUSSION THREAD

**Discussion by mcarans** [flakeheaven/flakeheaven#163] [General]
My main reason for using flakeheaven was for the pyproject.toml support. Given that flake8 major version changes break flakeheaven and flake8 seems to be a project in a state of great change, I thought about a relatively  simple script to generate a flake8 configuration file from a pyproject.toml which would be followed by running flake8 directly. I checked to see if anyone else had done this and noone had (in my brief search). 

However, I stumbled across an interesting project which was started in February:

https://github.com/john-hen/Flake8-pyproject
"Flake8 plug-in loading the configuration from pyproject.toml...

[Flake8](https://github.com/PyCQA/flake8) cannot be configured via pyproject.toml, even though virtually all other Python dev tools have adopted it as the central location for project configuration. The discussion of the original proposal (https://github.com/PyCQA/flake8/issues/234) was closed as "too heated", subsequent feature and pull requests were marked as "spam" (https://github.com/PyCQA/flake8/pull/1332, https://github.com/PyCQA/flake8/issues/1421, https://github.com/PyCQA/flake8/issues/1431, https://github.com/PyCQA/flake8/issues/1447, https://github.com/PyCQA/flake8/issues/1501).

Flake8-pyproject also has bad manners and force-feeds Flake8 the spam it so despises. It registers itself as a Flake8 plug-in to seamlessly load the configuration from pyproject.toml when you run the flake8 command."

I thought I'd share here in case of interest. I intend to give it a try in the new year to see if it meets my use case.

**Comment by pwoolvett** [comment]
agreed. I had also tried it and it seems to work fine.

On a side note, I'm starting to move my projects to ruff and have contributed a little bit to help improve it. I believe its the future.

If no new maintainer wants to take care of flakeheaven, we should archive and update the readme.

**Reply by mcarans** [reply]
I have been using flake8-pyproject for a while and am not using flakeheaven so I think it a good idea to ask for new maintainers.

I think you're right about ruff.
